File: astera/9/euler9.c
#include <stdio.h>

int main() {
  int a, b, c;
  int num = 1000;
  int result;

  for(a = 1; a < num; a++) {
    for(b = a; b < num; b++) {
      if((a*a + b*b) == ((num - a - b)*(num - a - b))) {
        c = (num - a - b);
        result = a*b*c;
        printf("%d\n", result);
        break;
      }
    }
  }
}

File: astera/9/euler9.rb
num = 1000

for a in (1..num)
  for b in (a..num)
    if a*a + b*b == (num - a - b)*(num - a - b)
      c = num - a - b
      puts "a = #{a}, b = #{b}, c = #{c} -- result = #{a*b*c}"
    end
  end
end
